app/src/main/java/com/rockspin/androiddevtest/APICommThread.java

```diff
@@ -0,0 +1,33 @@
+package com.rockspin.androiddevtest;
+
+import android.content.Intent;
+
+import java.util.List;
+
+import retrofit2.Response;
+
+/**
+ * Created by SG on 08.01.2017 г..
+ */
+public class APICommThread extends Thread
+{
+    @Override
```

Comment:
I think that relying on third party libs have their place depending on the library and it's maturity. The question becomes is your approach better than that developed in a community. I would never call threading simple, it introduces extra state into a program.